xarray/backends/zarr.py

```diff
@@ -1466,7 +1466,12 @@ def open_datatree(
             zarr_format=zarr_format,
             **kwargs,
         )
-        return datatree_from_dict_with_io_cleanup(groups_dict)
+        if group:
+            dt = datatree_from_dict_with_io_cleanup(groups_dict)
+            dt.encoding["source_group"] = group
```

Comment:
Thats right. I added the `encoding["source_group"]` here, but I was thinking more on creating a test for this group parameter. Do we need a test for this? If not, we might be ready

Reply:
We can leave this problem for a future PR.